cpp/src/arrow/compute/util.h

```diff
@@ -97,6 +97,9 @@ class ARROW_EXPORT TempVectorStack {
     return Status::OK();
   }
 
+  const int64_t buffer_size() const { return buffer_size_; }
+  static int64_t meta_size() { return kPadding + 2 * sizeof(uint64_t); }
```

Comment:
Can we provide `RequiredSize()`, `EstimateSize()` or something instead of providing this?